src/libs/ReportUtils.ts

```diff
@@ -8254,15 +8254,16 @@ function isExported(reportActions: OnyxEntry<ReportActions>) {
     return Object.values(reportActions).some((action) => ReportActionsUtils.isExportIntegrationAction(action));
 }
 
-function getApprovalChain(policy: OnyxEntry<Policy>, employeeAccountID: number, reportTotal: number): string[] {
+function getApprovalChain(policy: OnyxEntry<Policy>, expenseReport: OnyxEntry<Report>): string[] {
     const approvalChain: string[] = [];
+    const reportTotal = expenseReport?.total ?? 0;
 
     // If the policy is not on advanced approval mode, we should not use the approval chain even if it exists.
     if (!PolicyUtils.isControlOnAdvancedApprovalMode(policy)) {
         return approvalChain;
     }
 
```

Comment:
Ooh we should probably be getting rule approvers here too & adding them to the approval chain like this:

`[...categoryApprovers, ...tagApprovers, ...baseApprovalChain]` (de-duped) 🤔

Reply:
I think we don't need to do that here because `getSubmitToEmail` will return the `submitsTo` as `categoryApprover` or `tagApprover` if it is matched, and then we get this user's `forwardsTo`.

Reply:
Hmm ok so `ReportUtils.getApprovalChain` gets called in 2 places so far:
1. IOU.approveMoneyRequest
    - Here we mainly check `isLastApprover` of the approval chain so it probably doesn't matter adding category/tag approvers to the front of the approval chain
    - We also call `getNextApproverAccountID` to optimistically figure out who the next manager should be
2. IOU.getNextApproverAccountID
    - This gets called from buildNextStep and other places (ex: ^)

I thinkkkkk we will see problems if we run some tests with multiple category/tag approvers, because we'll only try to calculate a category/tag approver when submitting, but not as the "next approver" - at least in the optimistic onyx data. (possibly even backend errors, but that's what i'm working on fixing)

Reply:
@nkdengineer can you test my theory ^ there? I can mention exact test steps if you need

Reply:
> I can mention exact test steps if you need

@Beamanator Yes, please provide me the exact test steps.

Reply:
I think these are the steps to follow and confirm:
1. Create a workspace in NewDot
2. Enable delay submission
3. Invite submitter, Approver A, Approver B
4. Set to Advanced Approval in OldDot to workspace owner
5. Set category approvals for Category A going to Approver A + Category B as Approver B
6. As submitter in NewDot, submit 2 expenses to the WS chat, one with Category A, one with Category B
7. Submit the report
8. Confirm first approver is Approver A
9. As Approver A, approve the report
10. Confirm the second approver is Approver B
11. As Approver B, approve the report
12. Confirm the third approver is workspace owner
13. As WS owner, approve and confirm that the report is final approved.

Reply:
@Beamanator While testing the steps below, in step 9 approver A can't approve the report. BE returns an error like this.


<img width="1512" alt="Screenshot 2024-10-31 at 23 36 49" src="https://github.com/user-attachments/assets/769a396b-9449-4714-b6d5-dc3a1d2439ff">

Reply:
Thanks @garrettmknight - yes those look like great reproduction steps!

@nkdengineer - oof that's actually an old error that we should be getting rid of in https://github.com/Expensify/Web-Expensify/pull/44194, once that's deployed to staging (hopefully today) - I can try to ping here once we're ready for that deploy, but it probably won't happen till later (PST time)

Reply:
@Beamanator After approver A approves the money request, we're waiting for payment from admin. 

> I think we don't need to do that here because getSubmitToEmail will return the submitsTo as categoryApprover or tagApprover if it is matched, and then we get this user's forwardsTo

And it's what I said here.

https://github.com/user-attachments/assets/f8bef823-7ee9-4f89-937e-f62e7df277f3

Reply:
So it looks like after Approver A, we're expecting Approver B, right? I don't quite understand your comment though, sorry - did I lead you down a wrong path somewhere? 😅

Reply:
@Beamanator The video below is what BE working on after the Approver A approves if this user doesn't have `forwardsTo` then the report is marked as approved.

Reply:
Hmmm ok i can test on my end - but can you just make sure this work for you, optimistically? (offline) If it works well here optimistically, we can get this moving & i can fix the backend later

Reply:
So offline, we're expecting that after approver A approves the expense, the next one will be approver B right? I have a concern with this except if we have some approvers match with category/tag rules like A, B, C, and D. After the approvers A, and B approve the report, we don't have a way to detect the list of previous approves of the expense report.

Reply:
Yes exactly! And good call, that's an important consideration but I am ok with handling that in a separate PR - I just implemented this logic in the backend (not deployed yet) so i can help get this in the front end after this

Reply:
Then the next step is I will handle `getApproveChain` logic here or in a separate PR.

Reply:
Hmm so the issue this PR is fixing is purely related to "Submitting" so IMO in this PR we don't need to update the getApprovalChain logic - that can be a fast follow-up

Reply:
@Beamanator Got it, please check again. If it doesn't have any other problem. I think we're ready to merge this PR. Then we can create a follow-up later.

Reply:
Will do - resolving this thread too